src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp

```diff
@@ -67,6 +67,10 @@ namespace
             m_rendering_threads = 0;    // 0 = as many as there are logical cores
             m_low_priority_mode = true;
             m_use_max_procedural_maps = false;
+
+            const int log_mode = load_system_setting(L"LogOpenMode", static_cast<int>(DialogLogMode::Always));
```

Comment:
What happens if you remove the static cast and convert `log_mode` to `DialogLogMode`? What about `load_system_setting<DialogLogMode>()`?

Reply:
Cool! I could avoid extra assignment.

Reply:
ah, no.. looks like `T from_string(const std::string& s)` can't accept DialogLogMode.
Should I make `m_settings.m_log_open_mode` an int?

Reply:
I was suspecting that it wouldn't work. Let's keep it as you did (without the superfluous casts, if any).